hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientExecutorServiceProxy.java

```diff
@@ -402,20 +455,26 @@ private Object getTaskPartitionKey(Object task) {
         return checkSync(f, uuid, partitionId, false, defaultValue);
     }
 
-    private <T> Future<T> submitToKeyOwnerInternal(Data task, Object key, ExecutionCallback<T> callback) {
+    private <T> Future<T> submitToKeyOwnerInternal(@Nonnull Data task,
+                                                   @Nonnull Object key,
+                                                   @Nullable ExecutionCallback<T> callback) {
         checkNotNull(task, "task should not be null");
+        checkNotNull(key, "key should not be null");
         String uuid = getUUID();
         int partitionId = getPartitionId(key);
         ClientMessage request = ExecutorServiceSubmitToPartitionCodec.encodeRequest(name, uuid, task, partitionId);
         ClientInvocationFuture f = invokeOnPartitionOwner(request, partitionId);
 
-        ClientDelegatingFuture<T> delegatingFuture = new ClientDelegatingFuture<T>(f, getSerializationService(),
+        ClientDelegatingFuture<T> delegatingFuture = new ClientDelegatingFuture<>(f, getSerializationService(),
                 message -> ExecutorServiceSubmitToPartitionCodec.decodeResponse(message).response);
-        delegatingFuture.andThen(callback);
+        if (callback != null) {
```

Comment:
This fix and similar ones below are aligning behaviour with member-side where the callback is ignored if it is `null`.